register custom op

liangel-02 opened the pull request:
**Summary**

Adding custom op registration for `_varlen_attn()`. 

**Test plan**

Added `test_custom_op_registration` that uses logging mode to verify that `_varlen_attn()` was called and tests with `torch.compile`. Verified that the test fails when registration line is commented out

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #164950
* #164504
* __->__ #164503
* #164502

liangel-02 committed:
register custom op

[ghstack-poisoned]

liangel-02 committed:
Update on "register custom op"




[ghstack-poisoned]

liangel-02 committed:
Update on "register custom op"




[ghstack-poisoned]

jbschlosser reviewed (commented):
might have missed this but is the op tested in torch.compile?

liangel-02 committed:
Update on "register custom op"

**Summary**

Adding custom op registration for `_varlen_attn()`. 

**Test plan**

Added `test_custom_op_registration_with_logging()` that uses logging mode to verify that `_varlen_attn()` was called. Verified that the test fails when registration line is commented out




[ghstack-poisoned]

jbschlosser commented on test/test_varlen_attention.py:
```diff
@@ -81,59 +120,116 @@ def create_variable_length_batch(shape: VarlenShape, device: torch.device, dtype
+        )
+
+        compiled_forward = torch.compile(
+            attention_block.forward_varlen, backend="eager"
```
I'd specify `fullgraph=True` here to ensure we're not supporting this by graph breaking

liangel-02 committed:
Update on "register custom op"

**Summary**

Adding custom op registration for `_varlen_attn()`. 

**Test plan**

Added `test_custom_op_registration` that uses logging mode to verify that `_varlen_attn()` was called and tests with `torch.compile`. Verified that the test fails when registration line is commented out




[ghstack-poisoned]

liangel-02 committed:
Update on "register custom op"

**Summary**

Adding custom op registration for `_varlen_attn()`. 

**Test plan**

Added `test_custom_op_registration` that uses logging mode to verify that `_varlen_attn()` was called and tests with `torch.compile`. Verified that the test fails when registration line is commented out




[ghstack-poisoned]

liangel-02 committed:
Update

[ghstack-poisoned]

liangel-02 committed:
Update

[ghstack-poisoned]

liangel-02 committed:
Update

[ghstack-poisoned]

liangel-02 committed:
Update on "register custom op"

**Summary**

Adding custom op registration for `_varlen_attn()`. 

**Test plan**

Added `test_custom_op_registration` that uses logging mode to verify that `_varlen_attn()` was called and tests with `torch.compile`. Verified that the test fails when registration line is commented out




[ghstack-poisoned]

liangel-02 committed:
Update on "register custom op"

**Summary**

Adding custom op registration for `_varlen_attn()`. 

**Test plan**

Added `test_custom_op_registration` that uses logging mode to verify that `_varlen_attn()` was called and tests with `torch.compile`. Verified that the test fails when registration line is commented out




[ghstack-poisoned]

liangel-02 committed:
Update on "register custom op"

**Summary**

Adding custom op registration for `_varlen_attn()`. 

**Test plan**

Added `test_custom_op_registration` that uses logging mode to verify that `_varlen_attn()` was called and tests with `torch.compile`. Verified that the test fails when registration line is commented out




[ghstack-poisoned]

liangel-02 committed:
Update on "register custom op"

**Summary**

Adding custom op registration for `_varlen_attn()`. 

**Test plan**

Added `test_custom_op_registration` that uses logging mode to verify that `_varlen_attn()` was called and tests with `torch.compile`. Verified that the test fails when registration line is commented out




[ghstack-poisoned]

liangel-02 committed:
Update on "register custom op"

**Summary**

Adding custom op registration for `_varlen_attn()`. 

**Test plan**

Added `test_custom_op_registration` that uses logging mode to verify that `_varlen_attn()` was called and tests with `torch.compile`. Verified that the test fails when registration line is commented out




[ghstack-poisoned]

liangel-02 committed:
Update on "register custom op"

**Summary**

Adding custom op registration for `_varlen_attn()`. 

**Test plan**

Added `test_custom_op_registration` that uses logging mode to verify that `_varlen_attn()` was called and tests with `torch.compile`. Verified that the test fails when registration line is commented out




[ghstack-poisoned]

liangel-02 committed:
Update on "register custom op"

**Summary**

Adding custom op registration for `_varlen_attn()`. 

**Test plan**

Added `test_custom_op_registration` that uses logging mode to verify that `_varlen_attn()` was called and tests with `torch.compile`. Verified that the test fails when registration line is commented out




[ghstack-poisoned]
